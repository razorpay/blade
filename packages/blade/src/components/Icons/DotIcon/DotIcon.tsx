import { Svg, Circle } from '../_Svg';
import type { IconComponent } from '..';
import useIconProps from '../useIconProps';

const DotIcon: IconComponent = ({ size, color, ...styledProps }) => {
  const { height, width, iconColor } = useIconProps({ size, color });

  return (
    <Svg {...styledProps} width={width} height={height} viewBox="0 0 24 24" fill="none">
      <Circle cx="12" cy="12" r="8" fill={iconColor} />
    </Svg>
  );
};

export default DotIcon;
