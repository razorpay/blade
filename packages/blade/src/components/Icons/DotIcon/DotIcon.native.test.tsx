import DotIcon from '.';
import renderWithTheme from '~utils/testing/renderWithTheme.native';

describe('<DotIcon />', () => {
  it('should render DotIcon', () => {
    const renderTree = renderWithTheme(
      <DotIcon color="feedback.icon.neutral.intense" size="large" />,
    ).toJSON();
    expect(renderTree).toMatchSnapshot();
  });
});
