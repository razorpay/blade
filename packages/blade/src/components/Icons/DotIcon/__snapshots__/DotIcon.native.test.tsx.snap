// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DotIcon /> should render DotIcon 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSVGSvgView
    accessibilityElementsHidden={true}
    align="xMidYMid"
    bbHeight="20px"
    bbWidth="20px"
    data-blade-component="icon"
    fill="none"
    focusable={false}
    height="20px"
    importantForAccessibility="no-hide-descendants"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          {},
        ],
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width="20px"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGCircle
        cx="12"
        cy="12"
        fill={
          {
            "payload": 4280563015,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
        r="8"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
